File: isegoria/src/main/java/com/eulersbridge/isegoria/models/Ticket.java
package com.eulersbridge.isegoria.models;

import android.text.TextUtils;

import com.squareup.moshi.Json;

public class Ticket {

    @Json(name = "ticketId")
    public long id;

    public long electionId;

    public String name;
    public String information;

    private String colour;
    public String code;
    public String logo;

    @Json(name = "numberOfSupporters")
    public long supporterCount;

    public String getColour() {
        if (!TextUtils.isEmpty(colour)) {
            return colour;

        } else {
            return "#000000";
        }
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/Ticket.kt
package com.eulersbridge.isegoria.network.api.models

import com.squareup.moshi.Json

data class Ticket (

    @field:Json(name = "ticketId")
    val id: Long = 0,

    val electionId: Long = 0,

    val name: String?,
    val information: String?,

    private val colour: String?,
    val code: String?,
    val logo: String?,

    @field:Json(name = "numberOfSupporters")
    val supporterCount: Long = 0

) {
    fun getColour()
        = colour?.takeUnless { it.isBlank() } ?: "#000000"
}
